> I'm not sure the stress test shown would reveal the flake. I've tried to run it the same way locally without this patch and did not see it flaking:

@RomanBednar sorry,  I paste a wrong command in description, please try the following command: 

```
(⎈|kind-kind:N/A)➜  kubernetes git:(master) go test -c -race pkg/registry/core/persistentvolume/storage/storage.go pkg/registry/core/persistentvolume/storage/storage_test.go -run ^TestUpdateStatus$
(⎈|kind-kind:N/A)➜  kubernetes git:(master) stress -p 10  ./storage.test -test.run ^TestUpdateStatus$

/var/folders/n6/33y75ytn36g38k2wnkb6lp8r0000gn/T/go-stress-20231123T212946-1738443952
W1123 21:29:50.011337   19919 logging.go:59] [core] [Channel #4 SubChannel #7] grpc: addrConn.createTransport failed to connect to {Addr: "localhost:51571", ServerName: "localhost:51571", }. Err: connection error: desc = "transport: Error while dialing: dial tcp 127.0.0.1:51571: connect: connection refused"
--- FAIL: TestUpdateStatus (0.93s)
    storage_test.go:215: unexpected object:   core.PersistentVolumeStatus{
          	Phase:                   "Bound",
          	Message:                 "",
          	Reason:                  "",
        - 	LastPhaseTransitionTime: s"2023-11-23 21:29:49 +0800 CST",
        + 	LastPhaseTransitionTime: s"2023-11-23 21:29:50 +0800 CST",
          }
FAIL


ERROR: exit status 1

5s: 16 runs so far, 1 failures (6.25%)
```

--- a/pkg/registry/core/persistentvolume/storage/storage_test.go
+++ b/pkg/registry/core/persistentvolume/storage/storage_test.go
@@ -17,12 +17,10 @@ limitations under the License.
 package storage
 
 import (
-	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	featuregatetesting "k8s.io/component-base/featuregate/testing"
-	"k8s.io/kubernetes/pkg/features"
-	"testing"
-
 	"context"
+	"testing"
+	"time"
+
 	"github.com/google/go-cmp/cmp"
 	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/api/resource"
@@ -35,7 +33,10 @@ import (
 	genericregistrytest "k8s.io/apiserver/pkg/registry/generic/testing"
 	"k8s.io/apiserver/pkg/registry/rest"
 	etcd3testing "k8s.io/apiserver/pkg/storage/etcd3/testing"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
 	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/features"
 	"k8s.io/kubernetes/pkg/registry/core/persistentvolume"
 	"k8s.io/kubernetes/pkg/registry/registrytest"
 )
@@ -190,13 +191,16 @@ func TestUpdateStatus(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 
+	// We need to set custom timestamp which is not the same as the one on existing PV
+	// - doing so will prevent timestamp update on phase change and custom one is used instead.
+	pvStartTimestamp = &metav1.Time{Time: pvStartTimestamp.Time.Add(time.Second)}
+
 	pvIn := &api.PersistentVolume{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "foo",
 		},
 		Status: api.PersistentVolumeStatus{
-			Phase: api.VolumeBound,
-			// Set the same timestamp as original PV so this won't get updated on phase change breaking DeepEqual() later in test.
+			Phase:                   api.VolumeBound,
 			LastPhaseTransitionTime: pvStartTimestamp,
 		},
 	}

